steps/assert_steps.py

```diff
@@ -13,7 +13,10 @@ def step_impl(context, field):
     match field:
         case 'repos':
             ui_repo_number = context.search.get_text(Locators.number_of_repos)
-            expression = parse("$.public_repos")
+            expression = parse(JsonLocators.number_of_repos_json)
+        case 'followers':
+            ui_repo_number = context.search.get_text(Locators.number_of_followers)
```

Comment:
You can call all component classes directly:  ui_repo_number = Search(context.driver).get_text(Locators.number_of_followers) 
It is not necessarily to create an extra variable in context obj that is working as a global value and can lead to unpredictable behavior in future. The purpose of context variables is to pass data from once step to another within scenario run.